### PR TITLE
tf.keras: Warn user when they mix up `sparse_categorical_crossentropy` and `categorical_crossentropy`

### PR DESCRIPTION
**Pull request by facaiy**
Sometimes we convert one-hot labels to integer labels, and forget to change the loss function `categorical_crossentropy` to `sparse_categorical_crossentropy`.

Because `categorical_crossentropy` doesn't check the shape of `target` and `output`, tf.keras works well except a totally wrong result.

The PR only checks the last dimension of static shape for efficient, perhaps we'd better to check whole shape with help of `tf.contrib.framework.with_same_shape`?

### CODE REVIEW

**Review by fchollet — CHANGES_REQUESTED** [review]
There are similar checks being run in `training_utils.check_loss_and_target_compatibility()`. Any extension of such checks should happen at that level.

**Comment by facaiy** [comment]
@fchollet Yes, `training_utils.check_loss_and_target_compatibility()` contains the same checks. Thank you for pointing out this.
By the way, I cannot figure out where the function has been used in `tf.keras`, so does we expect users do the check by themselves explicitly?
